dynamiqs/mesolve/rouchon.py

```diff
@@ -54,33 +54,34 @@ def Ms(self, Hnh: Tensor, fwd: bool = True) -> tuple(Tensor, Tensor):
     def T(self, R: Tensor) -> Tensor:
         # we normalize the map at each time step by inverting `R` using its Cholesky
         # decomposition `R = T @ T.mT`
-        # -> (b_H, 1, n, n)
+        # -> (b_H, 1, 1, n, n)
         return cholesky(R)[0]  # lower triangular
 
     def forward(self, t: float, rho: Tensor) -> Tensor:
         r"""Compute $\rho(t+dt)$ using a Rouchon method of order 1.
 
         Args:
             t: Time.
-            rho: Density matrix of shape `(b_H, b_rho, n, n)`.
+            rho: Density matrix of shape `(b_H, b_L, b_rho, n, n)`.
 
         Returns:
-            Density matrix at next time step, as tensor of shape `(b_H, b_rho, n, n)`.
+            Density matrix at next time step, as tensor of shape
+            `(b_H, b_L, b_rho, n, n)`.
         """
-        # rho: (b_H, b_rho, n, n) -> (b_H, b_rho, n, n)
+        # rho: (b_H, b_L, b_rho, n, n) -> (b_H, b_L, b_rho, n, n)
 
-        H = self.H(t)  # (b_H, 1, n, n)
-        Hnh = self.Hnh(H)  # (b_H, 1, n, n)
-        M0, M1s = self.Ms(Hnh)  # (b_H, 1, n, n), (1, len(L), n, n)
+        H = self.H(t)  # (b_H, 1, 1, n, n)
+        Hnh = self.Hnh(H)  # (b_H, 1, 1, n, n)
+        M0, M1s = self.Ms(Hnh)  # (b_H, 1, 1, n, n), (len(L), 1, b_L, 1, n, n)
```

Comment:
Same comment here. I think there are mistakes in the shapes. Can you check them?